sopel_modules/imdb/imdb.py

```diff
@@ -44,35 +51,74 @@ def imdb(bot, trigger):
+        params['y'] = yrm.group(1)
+        word = ' '.join(word.split()[:-1])
+
+    params['t'] = word
+    bot.say(run_omdb_query(params, bot.config.core.verify_ssl, bot.config.imdb.api_key, True))
 
+def run_omdb_query(params, verify_ssl, api_key, add_url=True):
+    uri = "http://www.omdbapi.com/"
+    if 'i' in params:
+        data = requests.get(uri, params={'i': params['i'], 'apikey': api_key}, timeout=30, verify=verify_ssl)
+    elif 'y' in params:
+        data = requests.get(uri, params={'t': params['t'], 'y': params['y'], 'apikey': api_key}, timeout=30, verify=verify_ssl)
+    else:
+        data = requests.get(uri, params={'t': params['t'], 'apikey': api_key}, timeout=30, verify=verify_ssl)
+
+    data = data.json()
     if data['Response'] == 'False':
         if 'Error' in data:
             message = '[Error] %s' % data['Error']
         else:
-            LOGGER.warning(
-                'Got an error from the OMDb api, search phrase was %s; data was %s',
-                word, str(data))
+            LOGGER.info(
+                'Got an error from the OMDb api, search parameters were %s; data was %s',
+                str(params), str(data))
             message = '[Error] Got an error from OMDbapi'
     else:
+        data = data.json()
         if data['Type'] == 'movie':
-            message = '[Movie] Title: ' + data['Title'] + \
-                      ' | Year: ' + data['Year'] + \
-                      ' | Rating: ' + data['imdbRating'] + \
-                      ' | Genre: ' + data['Genre'] + \
-                      ' | IMDB Link: http://imdb.com/title/' + data['imdbID']
+            message = '[Movie] Title: ' + data['Title']
         elif data['Type'] == 'series':
             message = '[Series] Title: ' + data['Title'] + \
-                      ' | Seasons: ' + data['totalSeasons'] + \
-                      ' | Year: ' + data['Year'] + \
-                      ' | Rating: ' + data['imdbRating'] + \
-                      ' | Genre: ' + data['Genre'] + \
-                      ' | IMDB Link: http://imdb.com/title/' + data['imdbID']
-        else:
-            return
-    bot.say(message)
+                      ' | Seasons: ' + data['totalSeasons']
+
+        message += ' | Year: ' + data['Year'] + \
+                  ' | Rating: ' + data['imdbRating']
+        for rating in data['Ratings']:
+            if rating['Source'] == 'Rotten Tomatoes':
+                message += ' | Tomatometer: ' + rating['Value']
+        
+        message += ' | Genre: ' + data['Genre'] + \
+                   ' | Plot: {}'
+
+        if add_url:
+            message += ' | IMDB Link: http://imdb.com/title/' + data['imdbID']
+
+        plot = data['Plot']
+        if len(message.format(plot)) > 300:
+            cliplen = 300 - (len(message) - 2 + 3) # remove {} add […]
+            plot = plot[:cliplen] + '[…]'
+        message = message.format(plot)
+    return message
+
+@sopel.module.rule('.*(imdb\.com\/title\/)(tt[0-9]+).*')
```

Comment:
Go ahead and import rule alongside commands and example and abbreviate this decorator to just @rule